widgetssdk/src/main/java/com/glia/widgets/view/head/ActivityWatcherForChatHead.kt

```diff
@@ -38,15 +51,86 @@ internal class ActivityWatcherForChatHead(
 
     override fun onActivityPaused(activity: Activity) {
         controller.onActivityPaused()
+        removeChatHeadViewIfPresent()
         resumedActivity.clear()
     }
 
+    private fun createBubble(activity: Activity) {
+        val bubble = ChatHeadLayout(activity.baseContext)
+        bubble.layoutParams = ConstraintLayout.LayoutParams(
+            ViewGroup.LayoutParams.MATCH_PARENT,
+            ViewGroup.LayoutParams.MATCH_PARENT
+        )
+        bubble.setNavigationCallback(
+            object : ChatHeadLayout.NavigationCallback {
+                override fun onNavigateToEndScreenSharing() {
+                    navigateToEndScreenSharing(resumedActivity.get())
+                }
+
+                override fun onNavigateToChat() {
+                    navigateToChat(resumedActivity.get())
+                }
+
+                override fun onNavigateToCall() {
+                    navigateToCall(resumedActivity.get())
+                }
+            }
+        )
+        bubble.visibility = View.VISIBLE
+        this.bubble = WeakReference(bubble)
+    }
+
+    override fun addChatHeadViewIfAbsent() {
+        val viewName = fetchGliaOrRootView()?.javaClass?.simpleName
+        if (hasGliaView(resumedActivity.get()) || !controller.shouldShowBubble(viewName)) return
+        (fetchGliaOrRootView() as? ViewGroup)?.let { viewGroup ->
+            if (!viewGroup.hasChildOfType(ChatHeadLayout::class.java))
+                try {
+                    resumedActivity.get()?.let { activity ->
+                        createBubble(activity)
+                        Logger.d(TAG, "Adding application-only bubble")
+                        resumedActivity.get()?.runOnUiThread { viewGroup.addView(bubble.get()) }
+                    }
+                } catch (e: IllegalStateException) {
+                    Log.d(TAG, "Cannot add bubble: $e")
+                }
+        }
+    }
+
+    override fun removeChatHeadViewIfPresent() {
+        Logger.d(TAG, "Removing application-only bubble")
+        (fetchGliaOrRootView() as? ViewGroup)?.removeView(bubble.get())
+        bubble.clear()
+    }
+
+    private fun hasGliaView(activity: Activity?): Boolean {
+        var gliaView: View? = null
+        activity?.let {
+            gliaView = it.findViewById(R.id.call_view)
+                ?: it.findViewById<FilePreviewView>(R.id.file_preview_view)
+                        ?: it.findViewById<ChatView>(R.id.chat_view)
+                        ?: it.findViewById<EndScreenSharingView>(R.id.screen_sharing_screen_view)
+        }
+        return gliaView != null
+    }
+
+    private fun ViewGroup.hasChildOfType(clazz: Class<*>): Boolean {
+        for (i in 0 until childCount) {
+            val child = getChildAt(i)
+            if (clazz.isInstance(child)) {
+                return true
+            }
+        }
+        return false
+    }
```

Comment:
NIT: I think it could be moved to `ViewExtentions.kt`

Reply:
Done. f82ece80e3aa6fe3ca03d67bc53b38c5ae726f7e